src/tests/mod.rs

```diff
@@ -47,28 +50,46 @@ fn initialize() {
 /// * dest: Path to destination
 ///
 /// This method causes side effects - writes new files to a directory
-fn copy_folder(src: &Path, dst: &Path) {
-  for entry in fs::read_dir(src).unwrap() {
+fn copy_folder_to_temp_dir(src: &PathBuf) -> TempDir {
+  // Copy the test scenario to temporary directory
+  let temp_dir = TempDir::new_in(".", "tmp_test").unwrap();
+  let temp_dir_path = &temp_dir.path();
+  for entry in fs::read_dir(src.as_path()).unwrap() {
     let entry = entry.unwrap();
     let path = entry.path();
     if path.is_file() {
       _ = fs::copy(
         path.to_str().unwrap(),
-        dst.join(path.file_name().unwrap()).to_str().unwrap(),
+        temp_dir_path
+          .join(path.file_name().unwrap())
+          .to_str()
+          .unwrap(),
       )
       .unwrap();
     }
   }
+  temp_dir
 }
 
 /// Checks if the file updates returned by piranha are as expected.
-fn check_result(path_to_codebase: &Path, path_to_expected: &Path) {
+fn execute_piranha_and_check_result(
+  piranha_arguments: &PiranhaArguments, path_to_expected: &Path, files_changed: usize,
+  ignore_whitespace: bool,
+) {
+  let path_to_codebase = Path::new(piranha_arguments.path_to_codebase());
+
+  let output_summaries = execute_piranha(piranha_arguments);
+
+  assert!(output_summaries.iter().any(|x| !x.rewrites().is_empty()));
+
+  assert_eq!(output_summaries.len(), files_changed);
+
   let mut all_files_match = true;
 
   let count_files = |path: &Path| {
     let mut count = 0;
-    for dir_entry in fs::read_dir(path).unwrap() {
-      if dir_entry.unwrap().path().is_file() {
+    for dir_entry in fs::read_dir(path).unwrap().flatten() {
+      if dir_entry.path().is_file() && ".placeholder" != dir_entry.file_name().to_str().unwrap() {
```

Comment:
What is `".placeholder"`? I am assuming that we add these to empty output directories so that git allows to commit the directory? Is that true? Either way this could use a comment explaining.

Also, maybe worth it being a constant defined at the top of the file?